ui/main.qml

```diff
@@ -247,6 +248,76 @@ ApplicationWindow {
         property var appSettings
     }
 
+    DropArea {
+        id: dragTarget
+
+        signal droppedOnValidScreen(var drop)
+        property alias droppedUrls: rptDraggedPreviews.model
+        property bool enabled: containsDrag && loader.item &&
+            (
+                // in chat view in a one-to-one chat
+                (loader.item.currentView === 0 && chatsModel.activeChannel.chatType === Constants.chatTypeOneToOne) ||
+                // in timeline view
+                loader.item.currentView === 3
+            )
+
+        width: applicationWindow.width
+        height: applicationWindow.height
+
+        function cleanup() {
+            rptDraggedPreviews.model = []
+        }
+
+        onDropped: (drop) => {
+            if (enabled) {
+                droppedOnValidScreen(drop)
+            }
+            cleanup()
+        }
+        onEntered: {
+            // needed because drag.urls is not a normal js array
+            rptDraggedPreviews.model = drag.urls.filter(img => Utils.hasDragNDropImageExtension(img))
+        }
+        onPositionChanged: {
+            rptDraggedPreviews.x = drag.x
+            rptDraggedPreviews.y = drag.y
+        }
+        onExited: cleanup()
+        Rectangle {
+            id: dropRectangle
+
+            width: parent.width
+            height: parent.height
+            color: Style.current.transparent
+            opacity: 0.8
+
+            states: [
+                State {
+                    when: dragTarget.enabled
+                    PropertyChanges {
+                        target: dropRectangle
+                        color: Style.current.white
```

Comment:
Should this be `background` to work with the dark theme?

Reply:
Updated, great catch!